pycrdt_websocket/ystore.py

```diff
@@ -342,11 +341,10 @@ async def start(
         Arguments:
             task_status: The status to set when the task has started.
         """
-
         self.db_initialized = Event()
         if from_context_manager:
             assert self._task_group is not None
-            self._task_group.start_soon(self._init_db)
+            await self._init_db()
```

Comment:
@jzhang20133 is there a specific reason you removed the `start_soon` here? Was this causing a race condition? 

I ask because I think keeping this as part of the top level task_group is the "proper" way to structure concurrency here in an AnyIO world. Scheduling one-off tasks like this outside of the task_group might cause some issues, though I'm not experienced enough in AnyIO to know for sure.

Reply:
`self._task_group.start_soon(self._init_db)` will schedule this self._init_db task to run later in task group.  And `self._task_group.start_soon(self._init_db)` will not wait for _init_db to run and once task is scheduled, it moves to next line and finish this method.  For caller of ystore, when they call ystore.start(), after start() finish, it only make sure self._init_db is scheduled and there is no guarantee that it has been run. Hence I changes it to wait here so if caller also await ystore.start(), it can guarantee that self._init_db has been run and finished and after that, when other method of ystore is called, they don't throw runtimeError due to ystore is not running.

Reply:
I just did a test and it looks like there is no need to wait self._init_db here. As long as we await ystore.start call right after the ystore initialization in jupyter-collaboration and make sure self.db_initialized = Event() is called before any other methods of ystore are called, then we are good.

Reply:
for ystore class, after initialization, we also need to call start first before calling any other methods otherwise other methods will fail.

Reply:
I updated the PR accordingly now.

Reply:
Thanks, @jzhang20133